services/sfn/config/iam/recommended-policy-arn

```diff
@@ -0,0 +1 @@
+arn:aws:iam::aws:policy/AWSStepFunctionsFullAccess
```

Comment:
If I attach the above managed policy to my IAM role that I'm executing the service controller under, I'm getting the following error in my service controller logs when running the SFN smoke test:

```
AccessDeniedException: User: arn:aws:sts::750630568209:assumed-role/elasticache.services.k8s.aws.full/tmp-role-776639d8 is not authorized to perform: iam:PassRole on resource: arn:aws:iam::750630568209:role/ack-sfn-execution-role
```

Is there a need to also allow `iam:PassRole` permission?

Note that `arn:aws:sts::750630568209:assumed-role/elasticache.services.k8s.aws.full` is the IAM role I'm using for my service controller. Even though it's called "elasticache.services.k8s.aws.full", it has the `AWSStepFunctionsFullAccess` IAM managed policy associated with it.

Reply:
> Is there a need to also allow iam:PassRole permission?
Yes. Yes there is. 

Any advice on this? The only AWS managed policy that I'm aware of which contains sufficient permissions is `arn:aws:iam::aws:policy/IAMFullAccess`. I'd prefer that we don't suggest users sling that policy around too cavalierly. 

Maybe we could add a second file such as `services/sfn/config/iam/additional-policy` with a script/instructions to attach a policy with the iam:PassRole permissions?

Reply:
> > Is there a need to also allow iam:PassRole permission?
> > Yes. Yes there is.
> 
> Any advice on this? The only AWS managed policy that I'm aware of which contains sufficient permissions is `arn:aws:iam::aws:policy/IAMFullAccess`. I'd prefer that we don't suggest users sling that policy around too cavalierly.
> 
> Maybe we could add a second file such as `services/sfn/config/iam/additional-policy` with a script/instructions to attach a policy with the iam:PassRole permissions?

Hmm, yeah, that's a bit of a conundrum. I think adding a second file called `additional-policy` would be fine, yeah. If you wouldn't mind adding a quick note to the docs here:

https://github.com/aws/aws-controllers-k8s/blob/main/docs/contents/user-docs/authorization.md#configuring-aws-iam

stating that there may be an `additional-policy` file for the service, that would be great.

Thanks!

Reply:
I have added `services/sfn/config/iam/additional-policy` and mentioned it in `docs/contents/user-docs/authorization.md `. Give it another go, please.